pkg/agent/openflow/pipeline.go

```diff
@@ -2103,6 +2223,91 @@ func (c *client) serviceEndpointGroup(groupID binding.GroupIDType, withSessionAf
 	return group
 }
 
+// serviceSNATFlow generates the flow that handles the Service traffic which is from gateway.
+// For NodePort(externalTrafficPolicy is Cluster) or ClusterIP traffic from localhost or remote, do SNAT.
+// For NodePort(externalTrafficPolicy is Local) from remote, don't do SNAT, that means the traffic will reserve its source IP address.
+// For NodePort(externalTrafficPolicy is Local) from localhost, do SNAT.
+func (c *client) serviceSNATFlows(svcIP net.IP, svcPort uint16, protocol binding.Protocol, gatewayIP net.IP, isNodePortExternalLocal bool) []binding.Flow {
+	ipProtocol := getIPProtocol(svcIP)
+	serviceCtZone := ServiceCtZone
+	if ipProtocol == binding.ProtocolIPv6 {
+		serviceCtZone = ServiceCtZoneV6
+	}
+	table := c.pipeline[serviceSNATTable]
+	var flows []binding.Flow
+
+	flowBuilder := table.BuildFlow(priorityNormal).
```

Comment:
I feel it is more readable if put flow comments before each flow (rather than before the func).